### DISCUSSION TITLE
GitHub Actions' reusable workflows (from a custom path) have an incorrect `depName`

### DISCUSSION THREAD

**Discussion by jamietanna** [renovatebot/renovate#31988] [Request Help]
### How are you running Renovate?

Self-hosted, v37.425.1

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub.com

### Please tell us more about your question or problem

When scanning a repo with a [re-usable GitHub Actions workflow](https://docs.github.com/en/actions/sharing-automations/reusing-workflows) that includes a path, the `depName` doesn't get detected correctly.

For instance, via https://github.com/responsa/actions/blob/716b505130ed384785ac3364081ec11b1b8f1f3c/node.yaml#L1C1-L19C51 we can create a minimal repro.

Repro: https://github.com/jamietanna-testing/renovate-repro-github-actions-uses

As per the logs, we see that we get the following `depName`:

```
                 "depName": "responsa/actions",
```

However, we'd expect to see:

```
                 "depName": "responsa/actions/.github/workflows/build-node.yml"
```

### Logs (if relevant)

<details><summary>Logs</summary>

```
 INFO: Extracted dependencies (repository=local)
       "packageFiles": {
         "github-actions": [
           {
             "deps": [
               {
                 "depName": "responsa/actions",
                 "commitMessageTopic": "{{{depName}}} action",
                 "datasource": "github-tags",
                 "versioning": "docker",
                 "depType": "action",
                 "replaceString": "responsa/actions/.github/workflows/build-node.yml@main",
                 "autoReplaceStringTemplate": "{{depName}}/.github/workflows/build-node.yml@{{#if newDigest}}{{newDigest}}{{#if newValue}} # {{newValue}}{{/if}}{{/if}}{{#unless newDigest}}{{newValue}}{{/unless}}",
                 "currentValue": "main",
                 "skipReason": "github-token-required"
               }
             ],
             "packageFile": ".github/workflows/node.yml"
           }
         ]
       }

```

</details>

**Comment by rarkins** [comment]
I'm not sure I agree. For example if you imported a CDNJS library like this: `<script src=”https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.5.0/js/bootstrap.bundle.js”>` then the dependency is "twitter-bootstrap", not "twitter-bootstrap/js/bootstra.bundle.js" or something similar. For me the same goes for files within actions.

Another perspective: if you used multiple of those files from the same repo, and had an upgrade PR, you'd expect it to be a single PR with a title like "Update responsa/actions to v1.2.3"

**Reply by jamietanna** [reply]
Hmm. I'm in two minds.

I don't believe it's quite the same as the `twitter-bootstrap` example, but more like you pulling from a monorepo of packages? 

> Another perspective: if you used multiple of those files from the same repo, and had an upgrade PR, you'd expect it to be a single PR with a title like "Update responsa/actions to v1.2.3"

Yeah this makes me more agreeable with this POV.

However, there may be some cases where we want to split the PR by the actual Actions themselves i.e. if it's been a while between releases and some have breaking changes, and others are easier to pull in?  

The usage of a monorepo of actions reduces the overhead of multi-repo, but does cause some issues with then trying to version them in tandem

(Whether that's convincing enough to tweak this, I'm not sure)

**Reply by rarkins** [reply]
In the meantime you can potentially make use of this: https://github.com/renovatebot/renovate/pull/31925
